utils/utils.go

```diff
@@ -9,3 +11,13 @@ func GenerateRandomID() (string, error) {
 	u2, err := uuid.NewV4()
 	return u2.String(), err
 }
+
+func RandomPick(pickProbability float64) bool {
```

Comment:
Is there any value in injecting `rand` as a dependency as a second parameter of type interface to this function or by creating a `RandomPicker` struct with the dependency being a field and this function is a receiver method so that we can mock it for testing purposes? Is that overkill @SyntaxNode @guscarreon?